core/src/main/java/tc/oc/pgm/variables/VariablesModule.java

```diff
@@ -74,7 +75,16 @@ public static <T extends Filterable<?>> VariableCache<T> of(
   @Override
   public VariablesMatchModule createMatchModule(Match match) throws ModuleLoadException {
     for (VariableDefinition<?> varDef : this.variables) {
-      match.getFeatureContext().add(new Variable<>(varDef));
+      switch (varDef.getVariableType()) {
```

Comment:
```suggestion
      match.getFeatureContext().add(varDef.buildInstance());
```
buildInstance method can call the builder in type with itself as parameter